### DISCUSSION TITLE
Denormalisation of single-element lists for discriminator use

### DISCUSSION THREAD

**Discussion by buffyg** [pydantic/pydantic#5231] [Feature Request]
I've run into a situation with list members for parsing where that member is a list in the abstract schema that governs all possible result but is in some concrete API responses sub-typed to be effectively denormalised. What I want to be able to do is progressively parse down the type of the API response to work out the sub-typing that applies, then applying the most specific model. The immediate problem I've found is that discriminators seem to be implicitly scalar, even to the point that I can't, for lack of better terms, define the comparison to wrap some listiness into the comparison (I can't say that it's a Literal interned in a list, making said list either a conlist constrained to length 1 or saying that I only want to parse the first element, thereby eliminating iteration as opening an unbounded set can of worms).

I'm not asking because I want this problem or have any choice about changing what's effective the root schema for an API response, I'm asking because this is the hand I've been dealt. I'm consuming an major vendor's existing API, want to use Pydantic in the most straightforward way, and there's very little chance the vendor is going to agree that this is a bug they need to fix. Also not trying to implicitly shame anyone here, I don't think this kind of variable schema use with lists that can be scalar in sub-typed responses is obviously incorrect interface design, but maybe I'm missing references that designate this as breaking standards of practice,

**Comment by buffyg** [comment]
Anyone able to assist with this?

**Comment by chrisK824** [comment]
Can you please provide some piece of code as an example with what you are trying to achieve?
